rust/src/wilcoxon.rs

```diff
@@ -0,0 +1,479 @@
+                if group_char == 'X' {
+                    weight_x += i as f64 + 1.0;
+                } else if group_char == 'Y' {
+                    weight_y += i as f64 + 1.0;
+                }
+                //rank_frequencies.push(RankFreq {
+                //    rank: i as f64 + 1.0,
+                //    freq: 1,
+                //});
+                rank_frequencies.push(1.0);
+            } else {
+                frac_rank = calculate_frac_rank(i as f64 + 1.0, num_repeats);
+                ranks.push(frac_rank);
+                if group_char == 'X' {
+                    weight_x += frac_rank;
+                } else if group_char == 'Y' {
+                    weight_y += frac_rank
+                }
+                //rank_frequencies.push(RankFreq {
+                //    rank: frac_rank,
+                //    freq: num_repeats as usize,
+                //});
+                rank_frequencies.push(num_repeats);
+            }
+        } else if repeat_iter < num_repeats {
+            // Repeat case
+            ranks.push(frac_rank);
+            repeat_iter += 1.0;
+            if group_char == 'X' {
+                weight_x += frac_rank;
+            } else if group_char == 'Y' {
+                weight_y += frac_rank
+            }
+            if repeat_iter == num_repeats {
+                is_repeat = false;
+            }
+        } else {
+            //println!("i:{}", i);
+            ranks.push(i as f64 + 1.0);
+            repeat_iter = 1.0;
+            num_repeats = 1.0;
+            if group_char == 'X' {
+                weight_x += i as f64 + 1.0;
+            } else if group_char == 'Y' {
+                weight_y += i as f64 + 1.0;
+            }
+        }
+    }
+    //println!("rank_frequencies:{:?}", rank_frequencies);
+    //println!("xy:{:?}", xy);
+    //println!("ranks:{:?}", ranks);
+    //println!("weight_x:{}", weight_x);
+    //println!("weight_y:{}", weight_y);
+
+    //u_dash (calculated below) calculates the "W Statistic" in wilcox.test function in R
+
+    let u_y = weight_y - (group2.len() as f64 * (group2.len() as f64 + 1.0) / 2.0) as f64;
+    let u_dash_y = (u_y - (group1.len() * group2.len()) as f64).abs();
+    //println!("u_dash_y:{}", u_dash_y);
+
+    let u_x = weight_x - (group1.len() as f64 * (group1.len() as f64 + 1.0) / 2.0) as f64;
+    let u_dash_x = (u_x - (group1.len() * group2.len()) as f64).abs();
+    //println!("u_dash_x:{}", u_dash_x);
+
+    // Calculate test_statistic
+
+    //let t1 = weight_x - ((group1.len() as f64) * (group1.len() as f64 + 1.0)) / 2.0;
+    //let t2 = weight_y - ((group2.len() as f64) * (group2.len() as f64 + 1.0)) / 2.0;
+    //
+    //let mut test_statistic = t1;
+    //if t2 < t1 {
+    //    test_statistic = t2;
+    //}
+
+    //println!("test_statistic:{}", test_statistic);
+
+    if (group1.len() <= threshold && repeat_present == false)
+        || (group2.len() < threshold && repeat_present == false)
```

Comment:
Should it be `group1.len() < threshold`?

Reply:
I checked the source code of the R wilcox.test() it invokes the test when group1.len() <50 && group2.len() <50 && !ties . This has been updated in the rust source code also. This was also the source of the discrepancy in the example you have pointed out below. My latest commit resolves this discrepancy.